### PR TITLE
Refactor ahead of bulk npm

### PR DESCRIPTION
**Pull request by whit537**
Prereqs for #4488.

### Todo

- [x] change base to master after landing #4494

### CODE REVIEW

**Comment by whit537** [comment]
Ping @rohitpaulk. Ready for review! :-)

**Comment by whit537** [comment]
screen shot 2017-05-31 at 5 10 29 pm

https://twitter.com/Gratipay/status/870024324199088128 😞

File: gratipay/email.py [hunk 1]
```diff
@@ -86,9 +86,10 @@ def put(self, to, template, _user_initiated=True, **context):
                     raise Throttled()
 
 
-    def flush(self):
+    def flush(self, _flush_one=None):
         """Load messages queued for sending, and send them.
         """
+        flush_one = _flush_one or self._flush_one
```
Comment (rohitpaulk) [inline gratipay/email.py]: Hmm, why is this needed? We've got a console mailer hooked up in the test env, no?

Comment (rohitpaulk) [inline gratipay/email.py]: If the aim is to empty the email queue in tests, might be clear to use a separate 'email_queue.purge()' function. (Tests shouldn't have to be aware of the fact that we purge one email at a time)

**Comment by rohitpaulk** [comment]
Looks good to me, aside from [this one comment](https://github.com/gratipay/gratipay.com/pull/4495#discussion_r119508819)

File: gratipay/email.py (hunk 1 above)
Comment (whit537) [inline gratipay/email.py]: Better in c74a4137e448ab0ce31d7c561be871829702ce2b?
